crates/bevy_asset/src/io/file_asset_io.rs

```diff
@@ -49,7 +56,12 @@ impl FileAssetIo {
         file_asset_io
     }
 
-    pub fn get_root_path() -> PathBuf {
+    /// Returns the path of the parent directory of the appplication's executable.
+    ///
+    /// If the `CARGO_MANIFEST_DIR` environment variable is defined, which is set by cargo when
+    /// the app is run through it, the root path will be the crate's root instead, or whatever
+    /// path was passed to that variable.
+    pub fn get_exe_root() -> PathBuf {
```

Comment:
This doesn't actually get the "exe_root" when CARGO_MANIFEST_DIR is defined. I don't think this is a good name for the function. `get_root_path` is closer to being correct, but also isn't quite right (because the root path is actually `get_root_path` + `path`).

I'm open to suggestions here. Maybe `get_base_path`?

Reply:
I agree with `get_base_path`. My line of thinking with renaming the function was exactly because the actual root path is `{base}/assets`, and I settled on `get_exe_root` because I was thinking in terms of the working directory for the exe, when technically the exe root is `target/(debug|release)`.